### PR TITLE
Adds a target for installing terragrunt

### PR DESCRIPTION
**Pull request by lorengordon**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -112,6 +112,10 @@ terraform/install: | $(BIN_DIR) guard/program/jq
 	$(@D) --version
 	@ echo "[$@]: Completed successfully!"
 
+terragrunt/install: TERRAGRUNT_VERSION ?= latest
+terragrunt/install: | $(BIN_DIR) guard/program/jq
+	@ $(MAKE) install/gh-release/$(@D) FILENAME="$(BIN_DIR)/$(@D)" OWNER=gruntwork-io REPO=$(@D) VERSION=$(TERRAGRUNT_VERSION) QUERY='.name | endswith("$(OS)_$(ARCH)")'
+
```
Comment (kbalk) [inline Makefile:118]: I've seen a customized Makefile for the task I'm working on and it includes these additional lines for this target:

```
        chmod +x $(BIN_DIR)/$(@D)
        $(@D) --version
        @ echo "[$@]: Completed successfully!"
```

File: Makefile [hunk 2]
```diff
@@ -112,6 +112,10 @@ terraform/install: | $(BIN_DIR) guard/program/jq
 	$(@D) --version
 	@ echo "[$@]: Completed successfully!"
 
+terragrunt/install: TERRAGRUNT_VERSION ?= latest
+terragrunt/install: | $(BIN_DIR) guard/program/jq
+	@ $(MAKE) install/gh-release/$(@D) FILENAME="$(BIN_DIR)/$(@D)" OWNER=gruntwork-io REPO=$(@D) VERSION=$(TERRAGRUNT_VERSION) QUERY='.name | endswith("$(OS)_$(ARCH)")'
```
Comment (kbalk) [inline Makefile:117]: Perhaps insert before this line:

```
        @ echo "[$@]: Installing $(@D)..."
```

Comment (lorengordon) [reply Makefile:117]: That's done already in the target `install/gh-release/%`... for example:

```
$ make terragrunt/install
[install/gh-release/terragrunt]: Installing terragrunt...
curl --fail -sSL -H "Authorization: token $GITHUB_ACCESS_TOKEN" -o /home/loren/bin/terragrunt https://github.com/gruntwork-io/terragrunt/releases/download/v0.29.2/terragrunt_linux_amd64
chmod +x /home/loren/bin/terragrunt
terragrunt --version
terragrunt version v0.29.2
[install/gh-release/terragrunt]: Completed successfully!
```

File: Makefile (hunk 1 above)
Comment (lorengordon) [reply Makefile:118]: This is also done already, in the target `install/gh-release/%`...

**Review by kbalk — APPROVED** [review]

**Pull request merged**